motionplan/nearestNeighbor.go

```diff
@@ -138,43 +124,29 @@ func (nm *neighborManager) nnWorker(ctx context.Context, planOpts *plannerOption
 	var best node
 	bestDist := math.Inf(1)
 
-	for {
-		select {
-		case <-ctx.Done():
-			nm.neighbors <- nil
-			return
-		default:
+	for candidate := range nm.nnKeys {
```

Comment:
For our purposes, this framing of:
```
for candidate := range <channel> {
 ...
 <check context>
}
```
is fine. The producer is adding elements onto the channel as fast as it can. We never have to worry about blocking indefinitely for the next "candidate". But a more robust loop would look like:
```
for {
  select {
    case <-ctx.Done():
      double-break/return
    case candidate, ok := <-<channel>:
      if !ok {
        double-break/return
      }
      ...
}
```